src/Wallet.ts

```diff
@@ -0,0 +1,252 @@
+        chain_stats.funded_txo_count > 0 ||
+        chain_stats.spent_txo_count > 0 ||
+        mempool_stats.funded_txo_count > 0 ||
+        mempool_stats.spent_txo_count > 0;
+
+      const addressBalance =
+        chain_stats.funded_txo_sum - chain_stats.spent_txo_sum;
+
+      balance += addressBalance;
+      pendingBalance +=
+        mempool_stats.funded_txo_sum - mempool_stats.spent_txo_sum;
+
+      const addressMetadata = {
+        address: address!,
+        derivationPath: `${node.index}/${addressIndex}`,
+        publicKey: childNode.publicKey,
+      };
+
+      addresses.push(addressMetadata);
+
+      if (addressBalance > 0) {
+        addressesWithUtxo.push(addressMetadata);
+      }
+
+      if (!isUsedAddress && !nextUnusedAddress) {
+        nextUnusedAddress = addressMetadata;
+      }
+
+      if (!isUsedAddress) {
+        gap++;
+      } else {
+        gap = 0;
+      }
+
+      addressIndex++;
+    }
+
+    return {
+      balance,
+      pendingBalance,
+      addresses,
+      addressesWithUtxo,
+      nextUnusedAddress: nextUnusedAddress!,
+    };
+  };
+
+  fetchWalletStats = async (masterPublicKey: string): Promise<WalletStats> => {
+    const hdNode = bip32.fromBase58(masterPublicKey);
+    const [addressesData, changeAddressesData] = await Promise.all([
+      this.fetchAddressCollectionStats(hdNode.derive(0)),
+      this.fetchAddressCollectionStats(hdNode.derive(1)),
+    ]);
+
+    const balance = addressesData.balance + changeAddressesData.balance;
+    const pendingBalance =
+      addressesData.pendingBalance + changeAddressesData.pendingBalance;
+    const nextUnusedAddress = addressesData.nextUnusedAddress;
+    const nextUnusedChangeAddress = changeAddressesData.nextUnusedAddress;
+    const addresses = addressesData.addresses;
+    const changeAddresses = changeAddressesData.addresses;
+    const addressesWithUtxo = addressesData.addressesWithUtxo.concat(
+      changeAddressesData.addressesWithUtxo,
+    );
+
+    return {
+      balance,
+      pendingBalance,
+      nextUnusedAddress,
+      nextUnusedChangeAddress,
+      addresses,
+      changeAddresses,
+      addressesWithUtxo,
+    };
+  };
+
+  fetchUtxos = async (
+    addressesWithUtxo: AddressMetadata[],
+  ): Promise<UtxoWithMetadata[]> => {
+    const totalUtxos = [];
+
+    for (const addressWithUtxo of addressesWithUtxo) {
+      const utxos = await this.blockchainService.fetchAddressUtxos(
+        addressWithUtxo.address,
+      );
+
+      for (const utxo of utxos) {
+        totalUtxos.push({
+          ...utxo,
+          address: addressWithUtxo.address,
+          derivationPath: addressWithUtxo.derivationPath,
+        });
+      }
+    }
+
+    return totalUtxos;
+  };
+
+  private findPublicKeyForAddress = (
+    address: string,
+    addressesMetadata: AddressMetadata[],
+  ) =>
+    addressesMetadata.find(
+      (addressMetadata) => address === addressMetadata.address,
+    )?.publicKey;
+
+  createTransaction = async (
```

Comment:
The developer would call this method, passing values mostly retrieved from the other methods, to create a transaction.